install/kubernetes/helm/istio/templates/clusterrole.yaml

```diff
@@ -9,3 +9,34 @@ rules:
   - apiGroups: ["extensions", "apps"]
     resources: ["replicasets"]
     verbs: ["get", "list", "watch"]
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: istio-analyze
```

Comment:
doesn't galley already have all of this? I am not sure why we need to add this

Reply:
I see from the doc maybe it is useful, but we probably don't want the role binding?

Reply:
These are the goals I'm aiming at, and I'm very open to ideas on how to more elegantly hit them.

1. Have an Istio-maintained role so that it's straightforward to bind a user or service account that can run `istioctl analyze` without permission errors.
2. Have a canonical location documenting exactly what permissions `analyze` needs. (It's very close to what Galley uses, but not quite identical). 
3. Make sure that as we add analyzers that may depend on new  resources, Galley has the right permissions to run them when using `enableAnalysis`. For example, we added an analyzer that looks for valid references to k8s `secret` resources, but forgot to update Galley's permissions to include being able to read that. Having one role that's canonical for all analysis cases should make it easier to avoid that kind of mistake.

Reply:
I am still not understanding why the clusterrolebinding is needed as a helm template. Can we just mention in the docs that in cases where they need granular permissions, use the cluster role and bind to account they need to use for analyze?

Reply:
The binding is for the Galley status update case, so that the permissions used for analyze by Galley are consistent with the permissions used for analyze by istioctl. 

Alternatives I considered, but like less, are to either maintain separate roles for istioctl and galley (and update both when analyzers need new permissions), or to expect users to bind to the Galley role for the purpose of using istioctl.

Reply:
discussed offline. LGTM now.

Reply:
@howardjohn Does this reasoning make sense to you? Happy to discuss offline if it would help. If you agree this approach is OK, I need an env WG approver for this PR.

Reply:
What's the delta between the existing galley role and this new one?

Reply:
@ayj The Galley role has "get" in addition to "list" and "watch", but I'm not sure it even uses them.

The galley role additionally has some "update" permissions for `validatingwebhookconfigurations` and `deployments/finalizers`. And "update" permissions on the status subresource for Istio CRDs.

The galley role does *not* have permissions to read the `configmap` resource for mesh config. It gets that data via the file mount instead. It's also missing list/watch permissions for the `secrets` resource, which is used by an analyzer.

Reply:
Talked with @ayj some more and decided we can hold off on adding a special clusterrole or clusterrolebinding for now. The future maintenance burden of adding something new here is significant, and we don't know that it's even in demand. When permissions are missing, the error messages are very clear, and I expect it wouldn't take someone long to figure out what RBAC their user needs if they're running as a nonprivileged user. Given that, adding a maintenance burden in order to save a relatively small amount of user work is probably premature.